synapse/lib/storm.py

```diff
@@ -918,3 +914,64 @@ def getArgParser(self):
         pars = Cmd.getArgParser(self)
         pars.add_argument('delay', type=float, default=1, help='Delay in floating point seconds.')
         return pars
+
+class GraphCmd(Cmd):
+    '''
+    Generate a subgraph from the given input nodes and command line options.
+    '''
+    name = 'graph'
+
+    def getArgParser(self):
+
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('--degrees', type=int, default=1, help='How many degrees to graph out.')
+
+        pars.add_argument('--pivot', default=[], action='append', help='Specify a storm pivot for all nodes. (must quote)')
+        pars.add_argument('--filter', default=[], action='append', help='Specify a storm filter for all nodes. (must quote)')
+
+        pars.add_argument('--form-pivot', default=[], nargs=2, action='append', help='Specify a <form> <pivot> form specific pivot.')
+        pars.add_argument('--form-filter', default=[], nargs=2, action='append', help='Specify a <form> <filter> form specific filter.')
+
+        return pars
+
+    async def execStormCmd(self, runt, genr):
+
+        rules = {
+            'degrees': self.opts.degrees,
+
+            'pivots': [],
+            'filters': [],
+
+            'forms': {},
+        }
+
+        for pivo in self.opts.pivot:
+            rules['pivots'].append(pivo[1:-1])
+
+        for filt in self.opts.filter:
+            rules['filters'].append(filt[1:-1])
+
+        for name, pivo in self.opts.form_pivot:
```

Comment:
Is there any way to evaluate filter/pivot syntax before running the graph stuff?

Reply:
What happens if you give bad syntax?

Reply:
In this case, the full storm query parser was used to parse for tokenization in the first place.  So, if they were invalid, it would fail to parse and never make it to execution.